pkg/webhook/nad/mutator.go

```diff
@@ -152,41 +135,43 @@ func (m *Mutator) ensureLabels(nad *cniv1.NetworkAttachmentDefinition, oldConf,
 		}}, nil
 }
 
-// If the vlan or bridge name is changed, we need to tag the route annotation outdated
-func tagRouteOutdated(nad *cniv1.NetworkAttachmentDefinition, oldConf, newConf *utils.NetConf) (admission.Patch, error) {
-	if oldConf.BrName == newConf.BrName && oldConf.Vlan == newConf.Vlan {
+// If the vlan/route mode is changed, we need to tag the route annotation outdated
+func tagRouteOutdated(oldNad, newNad *cniv1.NetworkAttachmentDefinition, oldConf, newConf *utils.NetConf) (admission.Patch, error) {
+	if newConf.IsKubeOVNCNI() {
 		return nil, nil
 	}
-
-	annotations := nad.Annotations
-	if annotations == nil {
-		annotations = make(map[string]string)
+	// even when vlan is not changed, the route mode can be changed from `auto` to `static` or vice versa
+	if newNad.Annotations == nil || len(newNad.Annotations) == 0 {
```

Comment:
i assume this can only happen when a NAD has been created but controller has not yet discovered the route from completed job?